src/coreclr/vm/perfmap.cpp

```diff
@@ -276,10 +353,13 @@ void PerfMap::LogJITCompiledMethod(MethodDesc * pMethod, PCODE pCode, size_t cod
         SString line;
         line.Printf(FMT_CODE_ADDR " %x %s\n", pCode, codeSize, name.GetUTF8());
 
-        // Write the line.
-        if(s_Current != nullptr)
         {
-            s_Current->WriteLine(line);
+            CrstHolder ch(&(s_csPerfMap));
+
+            if(s_Current != nullptr)
+            {
+                s_Current->WriteLine(line);
+            }
         }
         PAL_PerfJitDump_LogMethod((void*)pCode, codeSize, name.GetUTF8(), nullptr, nullptr);
```

Comment:
Now that we can independently turn on and off perfmap and jitdump functionality, I think it would be worthwhile to look at a more holistic approach to locking.  Right now, both perfmap and jitdump both have their own locks, which means that in cases like this, if jitdump is enabled and perfmap is disabled, we must take the perfmap lock on every jit operation, just to discover that we aren't going to log to the perfmap.  Then we release the lock and have to acquire the jitdump lock to write the jit dump.  Perhaps we can merge these two locks together into one so that as long as one of them is enabled, we had to take the lock anyway?

Reply:
Yeah, I can use the perfmap lock for jitdump too